glean_parser/templates/kotlin.geckoview.jinja2

```diff
@@ -0,0 +1,34 @@
+// -*- mode: kotlin -*-
+
+/*
+ * AUTOGENERATED BY glean_parser.  DO NOT EDIT.
+ */
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+@file:Suppress("PackageNaming")
+package {{ namespace }}
+
+import {{ glean_namespace }}.private.HistogramBase // ktlint-disable import-ordering no-unused-imports
+
+/*
+ * This class performs the mapping between Gecko Histograms and Glean SDK
+ * metric types.
+ */
+internal object GleanGeckoHistogramMapping {
```

Comment:
So, I think this implementation still means that there is one mapping per library -- not Glean (app) wide.  I think that's fine here, just wanted to call that out.  It also means (I think) that only engine-gecko has access to this mapping, not Glean itself (unless it's passed into Glean somehow)...

Reply:
Yes, you are correct. Given the specifics of this, this is fine: there should be *one* GeckoView component embedding Gecko, not multiple per app. For context, we will create a new `Gecko.RuntimeTelemetry.Delegate` implementation in `engine-gecko-nightly`: this delegate will be the only one instance interacting with `GleanGeckoHistogramMapping`, as shown below:

```
// TODO: Change RemoveMe.Delegate to RuntimeTelemetry.Delegate once the
// GV API from Bug 1566367 is merged.
class GeckoAdapter : RemoveMe.Delegate {
    override fun onTelemetryReceived(metric: RemoveMe.Metric) {
        GleanGeckoHistogramMapping[metric.name]?.accumulateSamples(metric.values)
    }

}
```